src/vaadin-overlay.html

```diff
@@ -317,6 +326,7 @@
           /**
            * When set to true, the overlay is hidden. This also closes the overlay
            * immediately in case there is a closing animation in progress.
+           * @type {boolean!}
```

Comment:
Yes, using `@type {boolean!}` is intentional to match with `lib.dom.d.ts` which contains:

```ts
interface HTMLElement extends Element, DocumentAndElementEventHandlers, ElementCSSInlineStyle, ElementContentEditable, GlobalEventHandlers, HTMLOrSVGElement {
    // ...
    hidden: boolean;
    // ...
```

I don't think we want to have a default value present in `d.ts` file though.

Reply:
I meant making the property have a default `value: false` (not annotating it for d.ts somehow), because now when the initial value of `hidden` is `undefined` it doesn't really match the strict `boolean` type definition.

My thinking was that if you use the type defs and see the type maybe you expect the value to always be a strict boolean (either `true` or `false`), but if you don't explicitly set it for vaadin-overlay, it will instead be `undefined` which might be unexpected if the user is reading the value when it hasn't been changed.

This is also inconsistent with the native [HTMLElement.hidden](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/hidden) property. All HTML elements have `elem.hidden == false` by default but here because of the Polymer property definition of `hidden` without an explicit default `value` it is overriden to `undefined`.

This seems like a minor bug to me (missing default value of `false`), so maybe it would be ok to just give it `value: false`, then it would also initially match the strict boolean type. But this could be done outside of this PR.

And maybe put the `!` in front (similar to usage in list-mixin)
```suggestion
           * @type {!boolean}
```

Reply:
Actually the `!` is not needed here.
```suggestion
           * @type {boolean}
```
also gives:
`hidden: boolean;`

Reply:
When using TS I think without the default value being fixed, you could pass the `hidden` property to a method that only accepts `true` or `false` and breaks with `undefined` without getting a compile time TS error and it would break in runtime if that was executed with the default value of `undefined`.

Reply:
Changed to use `@type {boolean}`.